pkg/merge/merge_impl/group_by/group_by_stream_merge_rows.go

```diff
@@ -0,0 +1,165 @@
+type GroupByStreamMergeRows struct {
+	mu           sync.RWMutex
+	stmt         xxast.SelectStatement
+	queue        merge.PriorityQueue
+	currentRow   proto.Row
+	isFirstMerge bool
+}
+
+func NewGroupByStreamMergeRow(rows []*merge.MergeRows, stmt xxast.SelectStatement) *GroupByStreamMergeRows {
+	orderByItems := make([]merge.OrderByItem, 0)
+	for _, item := range stmt.OrderBy {
+		column := item.Alias
+		if column == "" {
+			column = item.Expr.(xxast.ColumnNameExpressionAtom)[0]
+		}
+		orderByItems = append(orderByItems, merge.OrderByItem{
+			Column: column,
+			Desc:   item.Desc,
+		})
+	}
+	s := &GroupByStreamMergeRows{
+		stmt:         stmt,
+		queue:        merge.NewPriorityQueue(rows, orderByItems),
+		isFirstMerge: true,
+	}
+	return s
+}
+
+func (s *GroupByStreamMergeRows) Next() proto.Row {
+	return s.merge()
+}
+
+func (s *GroupByStreamMergeRows) getAggrUnitMap(stmt xxast.SelectStatement) map[string]merge.AggrUnit {
+	aggrMap := make(map[string]merge.AggrUnit, 0)
+	for _, sel := range stmt.Select {
+		// todo 暂时用Mock方便测试，待改回
+		//if fun, ok := sel.(*xxast.SelectElementFunction); ok {
+		if fun, ok := sel.(*testdata.MockSelectElementFunction); ok {
+			columnName := fun.ToSelectString()
+			if fun.Alias() != "" {
+				columnName = fun.Alias()
+			}
+			// todo 暂时用Mock方便测试，待改回
+			//aggrMap[columnName] = s.getAggrFunctions(fun.Function().(*xxast.AggrFunction).Name())
+			aggrMap[columnName] = s.getAggrFunctions(xxast.AggrCount)
+		}
+	}
+	return aggrMap
+}
+
+func (s *GroupByStreamMergeRows) getAggrFunctions(aggrName string) merge.AggrUnit {
+	switch aggrName {
+	case xxast.AggrAvg:
+		return new(aggr_unit.AvgAggrUnit)
+	case xxast.AggrMax:
+		return new(aggr_unit.MaxAggrUnit)
+	case xxast.AggrMin:
+		return new(aggr_unit.MinAggrUnit)
+	case xxast.AggrSum:
+		return new(aggr_unit.SumAggrUnit)
+	case xxast.AggrCount:
+		return new(aggr_unit.CountAggrUnit)
+	default:
+		panic(fmt.Errorf("unsupported aggr source type: %s", aggrName))
+	}
+}
+
+func (s *GroupByStreamMergeRows) merge() proto.Row {
+	if s.queue.Len() == 0 {
+		return nil
+	}
+	if s.isFirstMerge {
+		s.currentRow = s.queue.Peek().(*merge.MergeRows).GetCurrentRow()
+		s.isFirstMerge = false
+	}
+	aggrMap := s.getAggrUnitMap(s.stmt)
+	currentRow := s.currentRow
+	currentGroupValue := NewGroupByValue(s.stmt.GroupBy, currentRow)
+	for {
+		if currentGroupValue.equals(s.currentRow) {
+			s.aggregate(&aggrMap, s.currentRow)
+		} else {
+			break
+		}
+		if !s.hasNext() {
+			break
+		}
+	}
+
+	// TODO 需要row提供Encode方法，才能将 GetData()方法的值进行修改
+	//return currentRow
+	//return proto.Row{}
+	age, _ := currentRow.GetColumnValue("age")
+	row := testdata.NewMockRow(gomock.NewController(nil))
+	row.EXPECT().GetColumnValue("age").Return(age, nil).AnyTimes()
+	row.EXPECT().GetColumnValue("count(score)").Return(aggrMap["count(score)"].GetResult(), nil).AnyTimes()
+	return row
+}
+
+// todo avg 函数待适配
+func (s *GroupByStreamMergeRows) aggregate(aggrMap *map[string]merge.AggrUnit, row proto.Row) {
```

Comment:
The `map` is already a pointer, use `map[string]merge.AggrUnit` directly, it's different from using `*[]SomeType` when appending items into slice.